script.recap/confupdate.py

```diff
@@ -0,0 +1,132 @@
+
+        # Now save the new file
+        dialogXmlFile = xbmcvfs.File(dialogXml, 'w')
+        dialogXmlFile.write(dialogXmlStr)
+        dialogXmlFile.close()
+
+    ##########################################################################
+    # UPDATES FOR DialogVideoInfo.xml
+    ##########################################################################
+    # Makes all the required changes to DialogVideoInfo.xml
+    def _updateDialogVideoInfo(self):
+        # Get the location of the information dialog XML file
+        dialogXml = os_path_join(self.confpath, 'DialogVideoInfo.xml')
+        log("DialogVideoInfo: Confluence dialog XML file: %s" % dialogXml)
+
+        # Make sure the file exists (It should always exist)
+        if not xbmcvfs.exists(dialogXml):
+            log("DialogVideoInfo: Unable to find the file DialogVideoInfo.xml, skipping file", xbmc.LOGERROR)
+            self.errorToLog = True
+            return
+
+        # Load the DialogVideoInfo.xml into a string
+        dialogXmlFile = xbmcvfs.File(dialogXml, 'r')
+        dialogXmlStr = dialogXmlFile.read()
+        dialogXmlFile.close()
+
+        # Now check to see if the skin file has already had the recap bits added
+        if 'script.recap' in dialogXmlStr:
+            # Already have recap referenced, so we do not want to do anything else
+            # to this file
+            log("DialogVideoInfo: Recap already referenced in %s, skipping file" % dialogXml, xbmc.LOGINFO)
+            self.errorToLog = True
+            return
+
+        # Now we need to add the button after the Final button
+        previousButton = '<label>13511</label>'
+
+        if previousButton not in dialogXmlStr:
+            # The file has had a standard component deleted, so quit
+            log("DialogVideoInfo: Could not find final button, skipping file", xbmc.LOGERROR)
+            self.errorToLog = True
+            return
+
+        # Check to make sure we use a unique ID value for the button
+        idOK = False
+        idval = 102
+        while not idOK:
+            idStr = "id=\"%d\"" % idval
+            if idStr not in dialogXmlStr:
+                idOK = True
+            else:
+                idval = idval + 1
+
+        # Now add the Recap button after the Final one
+        DIALOG_VIDEO_INFO_BUTTON = '''\n\t\t\t\t\t</control>\n\t\t\t\t\t<control type="button" id="%d">
+\t\t\t\t\t\t<description>Recap</description>
+\t\t\t\t\t\t<include>ButtonInfoDialogsCommonValues</include>
+\t\t\t\t\t\t<label>$ADDON[script.recap 32001]</label>
+\t\t\t\t\t\t<onclick>RunScript(script.recap)</onclick>
+\t\t\t\t\t\t<visible>System.HasAddon(script.recap) + [Container.Content(tvshows) | Container.Content(seasons) | Container.Content(episodes)]</visible>'''
+
+        insertTxt = previousButton + (DIALOG_VIDEO_INFO_BUTTON % idval)
+        dialogXmlStr = dialogXmlStr.replace(previousButton, insertTxt)
+
+        self._saveNewFile(dialogXml, dialogXmlStr)
```

Comment:
i reckon the wording in the wiki was poor at best (and have been updated since then),
but it always included this clause:
`Exceptions to this rule may be granted in specific cases only. Please contact Team XBMC's add-on repository maintainers via the add-on mailing list if your add-on needs access to such files.`

from the forum discussion, it's pretty clear both Martijn & tamland strongly opposed to the modification other addon's files.

please note that altering the default skin will fail on all platforms except windows, as it's either installed on a read-only partition (openelec) or requires root privileges to make such changes.